__tests__/unit/job-id.test.ts

```diff
@@ -0,0 +1,40 @@
+import {JobId, nullJobId} from '../../src/job-id'
+
+describe('JobId', () => {
+  it('should contain the Id of a Job', () => {
+    const jobId = new JobId(42)
+
+    expect(jobId.getId()).toBe(42)
+  })
+
+  it('should fail when using invalid job value', () => {
+    const negativeJobId = (): JobId => {
+      return new JobId(-2)
+    }
+    const NaNJobId = (): JobId => {
+      return new JobId(NaN)
+    }
+    const zeroJobId = (): JobId => {
+      return new JobId(0)
+    }
+
+    expect(negativeJobId).toThrowError()
+    expect(NaNJobId).toThrowError()
+    expect(zeroJobId).not.toThrowError()
+  })
```

Comment:
hi @yeraydavidrodriguez we should include somewhere a clarification that we had plans to use the job number 0 as a control job. It could be just a comment. Anyway, I'm not sure if in that case, we would not allow using the O. I suppose we would use it but it means something special. I mean, the app has to be able to create a job whose ID is 0, so it should not throw an exception.

Since we are not going to implement the "control job 0" as we mentioned [here](https://github.com/Nautilus-Cyberneering/git-queue/issues/38#issuecomment-1125893704) I would change the test to something like this:

```typescript
  it('should be an integer greater or equal to 0', () => {
    const negativeJobId = (): JobId => {
      return new JobId(-2)
    }

    // We have reserved the 0 for a special control job meaning queue start/allocation job.
    const zeroJobId = (): JobId => {
      return new JobId(0)
    }

    const NaNJobId = (): JobId => {
      return new JobId(NaN)
    }

    expect(negativeJobId).not.toThrowError()
    expect(negativeJobId).toThrowError()
    expect(NaNJobId).toThrowError()
```

Although I think it's better if we split the test:

```typescript
it('should be an integer greater or equal to 0', () => {
   // Test limits: 0 and any other positive integer
   // Mention that ID 0 could have an special meaning in the future.
}
it('should fail with invalid id values', () => {
   // Test negative and NaN
}
```